internal/reconciliations/external_authorizer/external_authorizer_test.go

```diff
@@ -0,0 +1,240 @@
+package external_authorizer_test
+
+import (
+	"context"
+	"fmt"
+	operatorv1alpha2 "github.com/kyma-project/istio/operator/api/v1alpha2"
+	"github.com/kyma-project/istio/operator/internal/reconciliations/external_authorizer"
+	networkingv1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	networkingv1beta1 "istio.io/client-go/pkg/apis/networking/v1beta1"
+	securityv1beta1 "istio.io/client-go/pkg/apis/security/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func createFakeClient(objects ...ctrlclient.Object) ctrlclient.Client {
+	err := operatorv1alpha2.AddToScheme(scheme.Scheme)
+	Expect(err).ShouldNot(HaveOccurred())
+	err = corev1.AddToScheme(scheme.Scheme)
+	Expect(err).ShouldNot(HaveOccurred())
+	err = networkingv1alpha3.AddToScheme(scheme.Scheme)
+	Expect(err).ShouldNot(HaveOccurred())
+	err = securityv1beta1.AddToScheme(scheme.Scheme)
+	Expect(err).ShouldNot(HaveOccurred())
+	err = networkingv1beta1.AddToScheme(scheme.Scheme)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	return fake.NewClientBuilder().WithScheme(scheme.Scheme).WithObjects(objects...).Build()
+}
+
+var _ = Describe("Reconciliation", func() {
+	const (
+		expectedName    = "expected-name"
+		expectedPort    = 1234
+		expectedService = "a.svc.cluster.local"
+	)
+
+	istioCR := operatorv1alpha2.Istio{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "default",
+			ResourceVersion: "1",
+			Annotations:     map[string]string{},
+			UID:             "1234",
+		},
+		Spec: operatorv1alpha2.IstioSpec{
+			Config: operatorv1alpha2.Config{
+				Authorizers: []*operatorv1alpha2.Authorizer{
+					{
+						Name:    expectedName,
+						Service: expectedService,
+						Port:    expectedPort,
+					},
+				},
+			},
+		},
+	}
+
+	Context("Duplicated authorization name", func() {
+		It("should cause a warning state", func() {
```

Comment:
We are not validating that the CR is in warning state, but only that an error is returned. So we might want to adapt the test description.